### PR TITLE
service-identity 24.1.0 (with updated deps, test, maintainer)

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

References:
- includes #6
- unblocks https://github.com/conda-forge/staged-recipes/pull/26617
  - unblocks https://github.com/conda-forge/hypercorn-feedstock/pull/44 

Changes:
- [x] rework `outputs`
- [x] remove bash build script 
- [x] add tests with coverage 
- [x] add bot inspection
- [x] add maintainer :wave:

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
@conda-forge/service_identity ready for review :tada:

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
@conda-forge/service_identity ready for review 🎉

**Comment by bollwyvl** [comment]
@conda-forge/service_identity @conda-forge/core ready for review 🎉

**Comment by bollwyvl** [comment]
@conda-forge/service_identity @conda-forge/core ready for review 🎉

**Comment by xhochy** [comment]
Explicit ping for @isuruf , otherwise I would merge once Azure CI is back alive.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -58,6 +110,6 @@ about:
   dev_url: https://github.com/pyca/service-identity
 
 extra:
-  feedstock-name: service_identity
```
Comment (isuruf) [inline recipe/meta.yaml:61]: Please keep this here

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (bollwyvl) [reply recipe/meta.yaml:61]: [8e0e65b](https://github.com/conda-forge/service_identity-feedstock/pull/7/commits/8e0e65bd042bcfe6c46292f477a1d5e0747e7596)

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
@conda-forge/service_identity ready for review 🎉

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
@conda-forge/service_identity @conda-forge/core ready for review 🎉

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
@conda-forge/service_identity @conda-forge/core ready for review 🎉

**Comment by h-vetinari** [comment]
Sorry for the long wait on this, and thanks for stepping up to help out with the feedstock!

**Pull request merged**